src/lib/components/Instances/InstanceTableRow.svelte

```diff
@@ -51,33 +45,53 @@
 		contenteditable="plaintext-only"
 		class="box-border p-3"
 		bind:innerText={localInstanceInput}
-		onblur={() => changeInstance({ ...instance, input: localInstanceInput })}
+		onblur={() => {
+			instance.input = localInstanceInput;
+			updateInstance({ ...instance, input: localInstanceInput });
+		}}
 		onkeydown={(event) => {
 			if (event.key === 'Enter' && (event.shiftKey || event.metaKey)) {
 				event.currentTarget.blur();
 			}
 		}}
 	/>
-	{#if project.show_labels}
-		<td
-			contenteditable="plaintext-only"
-			class="box-border p-3"
-			bind:innerText={localInstanceLabel}
-			onblur={() => changeInstance({ ...instance, label: localInstanceLabel })}
-			onkeydown={(event) => {
-				if (event.key === 'Enter' && (event.shiftKey || event.metaKey)) {
-					event.currentTarget.blur();
-				}
-			}}
-		/>
-	{/if}
+
 	<td class="p-3">
 		{#await prediction}
 			Loading...
 		{:then pred}
 			{pred?.prediction}
 		{/await}
 	</td>
+	{#if project.show_labels}
+		<td class="relative h-full">
+			<button
```

Comment:
I'm mid on this. I think it's useful but a bit ugly/in the way.

Reply:
I changed the style. Merging for now to avoid conflicts but happy to reverse.